rxbinding-support-recyclerview/src/androidTest/java/com/jakewharton/rxbinding/support/v7/widget/RxRecyclerViewTest.java

```diff
@@ -0,0 +1,203 @@
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import rx.Subscription;
+import rx.android.schedulers.AndroidSchedulers;
+
+import static android.support.test.espresso.action.ViewActions.swipeDown;
+import static android.support.test.espresso.action.ViewActions.swipeLeft;
+import static android.support.test.espresso.action.ViewActions.swipeRight;
+import static android.support.test.espresso.action.ViewActions.swipeUp;
+import static com.google.common.truth.Truth.assertThat;
+
+@RunWith(AndroidJUnit4.class)
+public class RxRecyclerViewTest {
+  @Rule public final ActivityTestRule<RxRecyclerViewTestActivity> activityRule =
+      new ActivityTestRule<>(RxRecyclerViewTestActivity.class);
+
+  private final Instrumentation instrumentation = InstrumentationRegistry.getInstrumentation();
+
+  private RecyclerView view;
+  private ViewInteraction interaction;
+  private ViewDirtyIdlingResource viewDirtyIdler;
+
+  @Before public void setUp() {
+    RxRecyclerViewTestActivity activity = activityRule.getActivity();
+    view = activity.recyclerView;
+    interaction = Espresso.onView(ViewMatchers.withId(android.R.id.primary));
+    viewDirtyIdler = new ViewDirtyIdlingResource(activity);
+    Espresso.registerIdlingResources(viewDirtyIdler);
+  }
+
+  @After public void tearDown() {
+    Espresso.unregisterIdlingResources(viewDirtyIdler);
+  }
+
+  @Test public void scrollEvents() {
+    RecordingObserver<RecyclerViewScrollEvent> o = new RecordingObserver<>();
+    Subscription subscription = RxRecyclerView.scrollEvents(view)
+        .subscribeOn(AndroidSchedulers.mainThread())
+        .subscribe(o);
+    o.assertNoMoreEvents();
+
+    instrumentation.runOnMainSync(new Runnable() {
+      @Override public void run() {
+        view.scrollBy(0, 50);
+      }
+    });
+    RecyclerViewScrollEvent event1 = o.takeNext();
+    assertThat(event1).isNotNull();
+    assertThat(event1.dy()).isEqualTo(50);
+
+    instrumentation.runOnMainSync(new Runnable() {
+      @Override public void run() {
+        view.scrollBy(0, 0);
+      }
+    });
+    o.assertNoMoreEvents();
+
+    instrumentation.runOnMainSync(new Runnable() {
+      @Override public void run() {
+        view.scrollBy(0, -50);
+      }
+    });
+    RecyclerViewScrollEvent event2 = o.takeNext();
+    assertThat(event2).isNotNull();
+    assertThat(event2.dy()).isEqualTo(-50);
+
+    // Back at position 0. Trying to scroll earlier shouldn't fire any events
+    instrumentation.runOnMainSync(new Runnable() {
+      @Override public void run() {
+        view.scrollBy(0, -50);
+      }
+    });
+    o.assertNoMoreEvents();
+
+    subscription.unsubscribe();
+
+    instrumentation.runOnMainSync(new Runnable() {
+      @Override public void run() {
+        view.scrollBy(0, 50);
+      }
+    });
+    o.assertNoMoreEvents();
+
+    instrumentation.runOnMainSync(new Runnable() {
```

Comment:
For some reason, splitting it up results in the horizontal swipe tests falling over on an `INJECT_EVENTS` security exception on swipe events. I thought Espresso didn't have this problem though, any thoughts?

I've seen some mention online about ghost keyboard issues people have had, and can confirm it does work fine if I run this in Genymotion (keyboard never shows up anyway).